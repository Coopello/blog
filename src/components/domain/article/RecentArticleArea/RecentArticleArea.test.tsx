import renderer from "react-test-renderer";
import type { Article } from "src/models/article";

import { RecentArticleArea } from "./";

const dummyNoImageArticle: Article = {
  id: "1",
  title: "タイトル",
  description: "記事の説明",
  content: "記事の本文",
  eyecatch: {
    url: "https://example.com/eyecatch.jpg",
    height: 200,
    width: 300,
  },
  tags: [
    { id: "hoge", name: "タグ1" },
    { id: "fuga", name: "タグ2" },
    { id: "piyo", name: "タグ3" },
  ],
  author: {
    id: "1",
    createdAt: "2022-01-01T00:00:00.000Z",
    updatedAt: "2022-01-02T00:00:00.000Z",
    publishedAt: "2022-01-03T00:00:00.000Z",
    revisedAt: "2022-01-04T00:00:00.000Z",
    name: "著者名",
    job: "職業",
    description: "著者の自己紹介",
    github: "example",
    twitter: "example",
    icon: {
      url: "https://example.com/icon.jpg",
      height: 216,
      width: 216,
    },
  },
};

describe("domain/article/RecentArticleArea", () => {
  it("Snap Shot", () => {
    const component = renderer.create(
      <RecentArticleArea
        articles={{
          contents: [dummyNoImageArticle],
          totalCount: 1,
        }}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
