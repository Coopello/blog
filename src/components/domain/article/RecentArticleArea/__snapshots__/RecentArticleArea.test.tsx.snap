// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`domain/article/RecentArticleArea Snap Shot 1`] = `
<section
  className="flex w-full flex-col"
  style={
    {
      "gap": "1rem",
    }
  }
>
  <h2
    className="border-custom-primary px-3 py-1 text-custom-black sm:px-4 sm:py-2"
    style={
      {
        "borderLeftWidth": 8,
        "fontSize": "1.5rem",
        "lineHeight": "2rem",
      }
    }
  >
    最近の記事
  </h2>
  <ul
    className="grid grid-cols-1 gap-6"
  >
    <li>
      <article>
        <button
          className="flex w-full items-center gap-4 rounded bg-custom-white p-4 text-left backdrop:cursor-pointer hover:opacity-60"
          onClick={[Function]}
        >
          <img
            alt="記事のイメージ画像"
            className="hidden rounded object-contain sm:inline-block"
            data-nimg="1"
            decoding="async"
            height={120}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=https%3A%2F%2Fexample.com%2Feyecatch.jpg&w=256&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fexample.com%2Feyecatch.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Feyecatch.jpg&w=256&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={120}
          />
          <div
            className="flex flex-1 flex-col gap-1 overflow-x-hidden"
          >
            <h2
              className="border-b pb-1 text-xl"
            >
              タイトル
            </h2>
            <ul
              className="hidden-scrollbar flex w-full overflow-y-clip overflow-x-scroll"
            >
              <li
                className="whitespace-nowrap"
              >
                <span
                  className="rounded-sm py-1 text-sm"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "color": "#5AC8D8",
                      "paddingLeft": 0,
                      "paddingRight": 8,
                    }
                  }
                >
                  #
                  タグ1
                </span>
              </li>
              <li
                className="whitespace-nowrap"
              >
                <span
                  className="rounded-sm py-1 text-sm"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "color": "#5AC8D8",
                      "paddingLeft": 0,
                      "paddingRight": 8,
                    }
                  }
                >
                  #
                  タグ2
                </span>
              </li>
              <li
                className="whitespace-nowrap"
              >
                <span
                  className="rounded-sm py-1 text-sm"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "color": "#5AC8D8",
                      "paddingLeft": 0,
                      "paddingRight": 8,
                    }
                  }
                >
                  #
                  タグ3
                </span>
              </li>
            </ul>
            <p
              className="w-full description text-sm text-custom-gray"
            >
              記事の説明
            </p>
          </div>
        </button>
      </article>
    </li>
  </ul>
</section>
`;
